tests/Functional/FunctionalTestCase.php

```diff
@@ -4,84 +4,71 @@
 
 namespace tests\Libero\ContentApiBundle\Functional;
 
-use DirectoryIterator;
 use InvalidArgumentException;
-use PHPUnit\Framework\TestCase;
-use Symfony\Component\Filesystem\Filesystem;
+use LogicException;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Bundle\FrameworkBundle\Test\TestContainer;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\ResettableContainerInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\KernelInterface;
 use tests\Libero\ContentApiBundle\Functional\App\Kernel;
 use function tests\Libero\ContentApiBundle\capture_output;
 
-abstract class FunctionalTestCase extends TestCase
+abstract class FunctionalTestCase extends KernelTestCase
 {
-    /** @var Filesystem */
-    private static $filesystem;
+    /** @var ContainerInterface */
```

Comment:
is worth of a comment on why we are not sharing a `Kernel` between tests, but we share a `Container` (likely performance)

Reply:
The container isn't shared.

Reply:
(Assuming `bootKernel` is called again. Symfony has `static::$container = null` in `ensureKernelShutdown()`, haven't added it here to the typehint can remain not-nullable. It does call `reset()` so the container itself is different.)